### PR TITLE
Database: update to EPSG v11.007, and other updates

### PR DESCRIPTION
**Pull request by rouault**
- Database: update to EPSG v11.007
- Operation Geographic3D Offset By Velocity Grid: use geodetic datum anchor epoch, now that NAD83(CSRS)vX datums have one since EPSG v11.007
- Recognize EPSG:1079 'New Zealand Deformation Model' and map it to +proj=defmodel, removing custom entries from other_transformation_custom.sql. That one is not strictly related to this EPSG update, but I just happend to notice there is now EPSG entries for the deformation model. @ccrook : I've noticed that the EPSG entries are from ITRF96 to NZGD2000 whereas the .json deformation models are in the reverse order (from NZGD200 to ITRF96). Thus in the mapping from the EPSG model to PROJ's +proj=defmodel I must apply the +inv modifier.

### CODE REVIEW

File: data/sql/proj_db_table_defs.sql
Comment (jjimenezshaw) [inline data/sql/proj_db_table_defs.sql:1]: You can remove line 1139 `        AND NEW.name != 'NAD83(CSRS)v2 to NAD83(CSRS)v3 (1)' -- duplicate entry in EPSG`

Comment (rouault) [reply data/sql/proj_db_table_defs.sql:1]: done. thanks

**Comment by rouault** [comment]
The backport to `9.4` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply b1927ea2... Database: avoid now useless exception for  'NAD83(CSRS)v2 to NAD83(CSRS)v3 (1)' duplicate entry
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
Auto-merging data/sql/metadata.sql
Auto-merging scripts/build_db.py
Auto-merging test/cli/testprojinfo_out.dist
Auto-merging test/unit/test_factory.cpp
[backport-4093-to-9.4 81acaeca] Database: update to EPSG v11.007
 Author: Even Rouault <even.rouault@spatialys.com>
 Date: Tue Mar 19 10:06:46 2024 +0100
 16 files changed, 129 insertions(+), 60 deletions(-)
[backport-4093-to-9.4 c370bf33] Operation Geographic3D Offset By Velocity Grid: use geodetic datum anchor epoch, now that NAD83(CSRS)vX datums have one since EPSG v11.007
 Author: Even Rouault <even.rouault@spatialys.com>
 Date: Tue Mar 19 10:14:01 2024 +0100
 1 file changed, 26 insertions(+), 31 deletions(-)
Auto-merging data/sql/proj_db_table_defs.sql
Auto-merging scripts/build_db.py
Auto-merging test/cli/testprojinfo_out.dist
[backport-4093-to-9.4 f306f1e9] Recognize EPSG:1079 'New Zealand Deformation Model' and map it to +proj=defmodel, removing custom entries from other_transformation_custom.sql
 Author: Even Rouault <even.rouault@spatialys.com>
 Date: Tue Mar 19 10:51:38 2024 +0100
 12 files changed, 222 insertions(+), 132 deletions(-)
Auto-merging data/sql/proj_db_table_defs.sql
CONFLICT (content): Merge conflict in data/sql/proj_db_table_defs.sql

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-9.4 9.4
# Navigate to the new working tree
cd .worktrees/backport-9.4
# Create a new branch
git switch --create backport-4093-to-9.4
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick f1b4e7237f4e97b5170c0767175d2bb79ce996a6,67f340230190c2cb8f49f5be577209767242fe25,7eb1c3a57edb9a80915f01219c7e4528ef5c1072,b1927ea29db725e1b4bc1dbb261efdebcea08c95
# Push it to GitHub
git push --set-upstream origin backport-4093-to-9.4
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-9.4
```
Then, create a pull request where the `base` branch is `9.4` and the `compare`/`head` branch is `backport-4093-to-9.4`.

**Comment by rouault** [comment]
failed to make its way to 9.4

**Pull request merged**